### DISCUSSION TITLE
Prevent charging on startup if SOC is => SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT

### DISCUSSION THREAD

**Discussion by ofernander** [Louisvdw/dbus-serialbattery#827] [Ideas]
I've been trying to implement this but my coding skills are subpar to say the least. I essentially want to prevent charging when the driver is restarted. It makes troubleshooting/fine tuning harder and needlessly charges my batteries. Right now whenever VenosOS is rebooted or you manually restart the driver the entire charging cycle starts from the top. 

I propose there should be an option to tie max voltage/charging current to SOC defined by SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT.  So before something like "def manage_charge_voltage_linear(self) -> None:" even starts it checks to make sure the SOC is above the threshold defined by the user. 

@mr-manuel  where could this be implemented do you think? I've been trying to work within this def "def manage_charge_voltage_linear(self) -> None:"  trying to add an if statement like "if utils.SOC_LEVEL_TO_RESET_VOLTAGE_LIMIT <= self.soc:
         return"
         
But I'm getting weird attribute errors from daly.py that I can't track down. 

Thanks for any input, really interested in this project but like I said my coding skills are a little lacking but trying to think of ways to improve and contribute to this project.

**Comment by mr-manuel** [comment]
This is not needed, since I planned to save some values to persist after a restart/reboot, like:
* Current charging mode
* Days since last SOC reset
Is there another value that you think it might be useful?

**Reply by ofernander** [reply]
Total cycles if pulled from the BMS. 

Interested in how you're going to implement these features, will it be in a nightly build soon? 

If you do add this feature having a button in the GUI that would trigger a manual bulk charge back to 100% or N% would be a good idea in case the user wants to charge batteries out of the normal cycle. Maybe to take advantage of extra sunshine or in my case right before leaving the dock for the day.

**Reply by mr-manuel** [reply]
Why total cycles? If they are pulled from the BMS there is no need to save it.,

**Reply by ofernander** [reply]
To keep a true record of cycles. If the BMS is reset or a new BMS is put into place that info is lost.

**Comment by mr-manuel** [comment]
For reference: https://github.com/Louisvdw/dbus-serialbattery/issues/840
